Engine/AlgorithmManager.cs

```diff
@@ -689,6 +689,14 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, ISynchronizer syn
                 results.SendStatusUpdate(AlgorithmStatus.Deleted);
             }
 
+            //Runtime Error.
+            if (_algorithm.Status == AlgorithmStatus.RuntimeError)
+            {
+                Log.Trace("AlgorithmManager.Run(): Algorithm RuntimeError...");
+                Log.Trace("RuntimeError:" + _algorithm.RunTimeError);
```

Comment:
Not sure we need `Log.Trace("RuntimeError:" + _algorithm.RunTimeError);` since it's being logged before? Or maybe above, when detecting the runtimeError, instead of `break` we could directly use `return` as done with all the other `AlgorithmStatus.RuntimeError` cases in this loop, keep it normalized 🤔

Reply:
Gotcha, I was little worried about line 176 in AlgorithmManager  breaking out of the loop after the error is posted and then it being set the complete again, but now that I took a deeper look at the sequence it makes sense that it would catch it on line 327 after processing all the data for that time slice.

Reply:
Yeah exactly 👍 because the status isn't being set before but by the AlgorithmManager at 329